modules/internal/ChapelBase.chpl

```diff
@@ -441,7 +441,7 @@ module ChapelBase {
   //
   // bitwise operations on primitive types
   //
-  inline proc ~(a: bool) return __primitive("u~", a);
+  inline proc ~(a: bool) { compilerError("~ is not suported on bool"); }
```

Comment:
On the phone just now we were talking about how the where clause didn't do the right thing as expected.  I forgot to ask the obvious follow-up question:  Is there some other (not-too-contorted) solution available to us that you can see?  In any case, I think the test you wrote should be updated (or cloned) to put the user overload into a sibling "library" module which I suspect would demonstrate the unintended behavior?